pkg/controllers/work/apply_controller_test.go

```diff
@@ -828,6 +828,73 @@ func TestGenerateWorkCondition(t *testing.T) {
 	}
 }
 
+func TestIsDataResource(t *testing.T) {
+	tests := map[string]struct {
+		gvr  schema.GroupVersionResource
+		want bool
+	}{
+		"Namespace resource": {
+			gvr: schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "namespaces",
+			},
+			want: true,
+		},
+		"Secret resource": {
+			gvr: schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "secrets",
+			},
+			want: true,
+		},
+		"ConfigMap resource": {
+			gvr: schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "configmaps",
+			},
+			want: true,
+		},
+		"Role resource": {
+			gvr:  utils.RoleGVR,
+			want: true,
+		},
+		"RoleBinding resource": {
+			gvr:  utils.RoleBindingGVR,
+			want: true,
+		},
+		"ClusterRole resource": {
+			gvr:  utils.ClusterRoleGVR,
+			want: true,
+		},
+		"ClusterRoleBinding resource": {
+			gvr:  utils.ClusterRoleBindingGVR,
+			want: true,
+		},
+		"Non-data resource (Pod)": {
+			gvr: schema.GroupVersionResource{
+				Group:    "",
+				Version:  "v1",
+				Resource: "pods",
+			},
+			want: false,
+		},
+		"Non-data resource (Service)": {
+			gvr:  utils.ServiceGVR,
+			want: false,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := isDataResource(tt.gvr)
+			assert.Equal(t, tt.want, got)
```

Comment:
```suggestion
			if got != tt.want {
			     t.Errorf("isDataResource() = %v, want %v", got, tt.want)
			}
```
can we use errorf instead? the assert will stop the test in the middle if one of them fail. Need to run multiple times to figure out all the test failures.

Reply:
gpt created this test :)